homeassistant/components/template/switch.py

```diff
@@ -184,13 +192,21 @@ def available(self) -> bool:
     async def async_turn_on(self, **kwargs):
         """Fire the on action."""
         await self._on_script.async_run(context=self._context)
+        if self._template is None:
+            self._state = True
+            self.async_schedule_update_ha_state()
```

Comment:
Since there is nothing to call in `async_update`, just call `self.async_write_ha_state()`.